extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientBase.java

```diff
@@ -77,6 +80,12 @@ private void configureSsl(RestClientBuilder builder) {
         if (maybeHostnameVerifier.isPresent()) {
             registerHostnameVerifier(maybeHostnameVerifier.get(), builder);
         }
+
+        // we need to push a disabled SSL context when SSL has been disabled
+        // because otherwise Apache HTTP Client will try to initialize one and will fail
+        if (ImageInfo.inImageRuntimeCode() && !SslContextConfiguration.isSslNativeEnabled()) {
```

Comment:
What does `ImageInfo.inImageRuntimeCode()` return if it's running on OpenJDK?

Reply:
It returns `false`. We use that in a couple more places in Quarkus as well (for example: https://github.com/quarkusio/quarkus/blob/f89650be537d9f26c54c06c960b3e258bac2fe74/core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java#L157)

Reply:
Fair enough, I was half expecting it to majorly fail if it wasn't on GraalVM JDK

Reply:
Under the hood it just checks for a system property, so it's not that smart :)